File: problems/392/solution.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i = 0;
        
        for (const char& ch: t) {
            if (i == size(s)) {
                break;
            }
            
            if (s[i] == ch) {
                i++;
            }
        }
        
        return i == size(s);
    }
};

File: problems/392/solution.go
package main
func isSubsequence(s string, t string) bool {
    // if len(s) > len(t) {
    //     return false
    // }
    
    // if len(s) == 0 {
    //     return true
    // }
    
    p := 0
    
    for _, ch := range t {
        if p == len(s) {
            return true
        }
        
        if ch == rune(s[p]) {
            p++
        }
    }
    
    return p == len(s)
}